velociraptor/catalogue/soap_catalogue.py

```diff
@@ -0,0 +1,266 @@
+import numpy as np
+import h5py
+import unyt
+
+from typing import List, Set
+from types import SimpleNamespace
+
+from velociraptor.catalogue.catalogue import Catalogue
+from velociraptor.catalogue.translator import VR_to_SOAP
+
+from functools import reduce
+
+from astropy.cosmology import wCDM, FlatLambdaCDM
+import unyt
+
+
+class CatalogueElement(object):
+    file_name: str
+    name: str
+
+    def __init__(self, file_name, name):
+        self.file_name = file_name
+        self.name = name
+
+
+class CatalogueDataset(CatalogueElement):
+
+    conversion_factor: unyt.unyt_quantity
+    value: unyt.unyt_array
+
+    def __init__(self, file_name, name, handle):
+        super().__init__(file_name, name)
+
+        self.conversion_factor = None
+        self._value = None
+        self._register_metadata(handle)
+
+    def _register_metadata(self, handle):
+        metadata = handle[self.name].attrs
+        factor = (
+            metadata["Conversion factor to CGS (including cosmological corrections)"][0]
+            * unyt.A ** metadata["U_I exponent"][0]
+            * unyt.cm ** metadata["U_L exponent"][0]
+            * unyt.g ** metadata["U_M exponent"][0]
+            * unyt.K ** metadata["U_T exponent"][0]
+            * unyt.s ** metadata["U_t exponent"][0]
+        )
+        self.conversion_factor = unyt.unyt_quantity(factor)
+        self.conversion_factor.convert_to_base("galactic")
+
+    def get_value(self, group):
+        if self._value is None:
+            with h5py.File(self.file_name, "r") as handle:
+                self._value = handle[self.name][:] * self.conversion_factor
+            self._value.name = self.name.replace("/", " ").replace("_", "").strip()
+        return self._value
+
+    def set_value(self, value, group):
+        self._value = value
+
+    def del_value(self, group):
+        del self._value
```

Comment:
The `group` thing makes sense to me now. When we access the `property`, we are actually doing
```
getattr(<CatalogueGroup instance>, "name")
```
which is then translated into
```
<property getter for "name">(<CatalogueGroup instance>)
  --> <CatalogueDataset instance>.get_value(<CatalogueGroup instance>)
  --> get_value(<CatalogueDataset instance "self">, <CatalogueGroup instance>)
```
That means we need the `get_value(self, group)` syntax, even though `group` is never used.

Reply:
I think I disagree about moving `_value` and `set_value`, `del_value` to `CatalogueElement`. `CatalogueGroup` also inherits from this class and those objects do not need this variable and these functions. So that would not be very clean.

To avoid code duplication, I could introduce another class that inherits from `CatalogueElement` and that serves as parent class for `CatalogueDataset` and `CatalogueDerivedDataset`, but that would likely introduce more code than it saves.